### DISCUSSION TITLE
Change viewport settings like `view_center_point` or `view_height`

### DISCUSSION THREAD

**Discussion by Jopie01** [mozman/ezdxf#1182] [Q&A]
Related to https://github.com/mozman/ezdxf/discussions/1181

I have a predefined DXF template with an empty modelspace, but with a paperspace layout with several viewports. There are also different layers and configured in a way that some of them are frozen in different viewports.

Now I open the DXF file, select modelspace and a layer and add then entities. Then I move to paperspace and because the drawing in modelspace is different, the viewport needs to be updated with then new positions from modelspace. I can calculate the boundingbox of the modelspace and based on that and the viewport size I want to change the `dxf.view_center_point` and the `dxf.view_height` but it doesn't work.

Did I miss something or isn't it possible? In the latter, then I need to create the viewports myself everytime. Not a huge issue, but if it's possible to update the viewport, users can make changes to the template rather then dig into the code.

**Comment by denebnet** [comment]
This is possible.  You have to address the right Viewport Entity: Paperspace Layouts have a Main Viewport, which is "paper", and other Viewport(s) are the windows into ModelSpace.
```
>>> import ezdxf
>>> dwg = ezdxf.readfile("vpt.dxf")
>>> dwg.layouts.names()
['Model', 'Sample Layout']

# address the Paperspace Layout
>>> psp = dwg.layouts.get('Sample Layout')
>>> psp
<ezdxf.layouts.layout.Paperspace object at 0x7fa08c94a120>

# we have two Viewport Entities...
>>> psp.viewports()
[<class 'ezdxf.entities.viewport.Viewport'> VIEWPORT(#D02), <class 'ezdxf.entities.viewport.Viewport'> VIEWPORT(#D0B)]

# the "main" is almost certainly the first Viewport...
>>> psp.main_viewport()
<class 'ezdxf.entities.viewport.Viewport'> VIEWPORT(#D02)

# address the one that is not the "Main"
>>> vp = psp.viewports()[1]

# returns a dict of ALL the viewport's dxf attribs - there are many ...
>>> vp.dxfattribs()
{ ... 'center': Vec3(7.5, 5.5, 0.0), 'width': 14.5, 'height': 10.5, 'status': 2, 'id': 2, 'view_center_point': Vec3(13958.65796264492, 9117.979969681584, 0.0), 'view_height': 563.25,... }

# we want to point this viewport at a specific location in Model Space:
>>> vp.dxf.view_center_point = (68.55,72.80)

# "scaling" the view in the viewport: How many units of Modelspace shall fit inside the Viewport's Height (on paper)...
>>> vp.dxf.view_height = 300
>>> dwg.save()
# Open that in CAD, the viewport's view will have shifted as specified.

```
You can determine the values needed for `view_center_point` and `view_height` by any convenient means.
